Zend/zend_inheritance.c

```diff
@@ -2043,8 +2044,12 @@ static void zend_do_traits_property_binding(zend_class_entry *ce, zend_class_ent
 
 			Z_TRY_ADDREF_P(prop_value);
 			doc_comment = property_info->doc_comment ? zend_string_copy(property_info->doc_comment) : NULL;
+			if (property_info->attributes) {
+				attributes = property_info->attributes;
+				GC_ADDREF(attributes);
+			}
```

Comment:
I don't think this code path is tested. You likely can't unconditionally addref here if the property is in SHM.